pkg/cached/cache_test.go

```diff
@@ -765,60 +765,61 @@ func TestReplaceableWithTransformer(t *testing.T) {
 func Example() {
 	// Merge Json is a replaceable cache, since we'll want it to
 	// change a few times.
-	mergeJson := cached.Replaceable[[]byte]{}
+	mergeJson := cached.R(cached.LastSuccess[[]byte])(&cached.Atomic[[]byte]{})
 
-	one := cached.NewStaticSource(func() cached.Result[[]byte] {
+	one := cached.Once(cached.DataFunc(func() cached.Result[[]byte] {
 		// This one is computed lazily, only when requested, and only once.
-		return cached.NewResultOK([]byte("one"), "one")
-	})
-	two := cached.NewSource(func() cached.Result[[]byte] {
+		return cached.NewResult([]byte("one"), "one")
+	}))
+	two := cached.DataFunc(func() cached.Result[[]byte] {
 		// This cache is re-computed every time.
-		return cached.NewResultOK([]byte("two"), "two")
+		return cached.NewResult([]byte("two"), "two")
 	})
 	// This cache is computed once, and is not lazy at all.
-	three := cached.NewResultOK([]byte("three"), "three")
+	three := cached.NewResult([]byte("three"), "three")
 
 	// This cache will allow us to replace a branch of the tree
 	// efficiently.
-	replaceable := cached.Replaceable[[]byte]{}
-	replaceable.Replace(cached.NewResultOK([]byte("four"), "four"))
+
+	replaceable := cached.R(cached.LastSuccess[[]byte])(&cached.Atomic[[]byte]{})
```

Comment:
Personally this feels like a very complicated way to accomplish what we want with replaceable, but will defer to @apelisse.

Reply:
Logically this usage makes sense, I'm just not sure about use case of having this kind of function since almost all our current use cases involve `cached.R` with LastSuccess and Atomic. Are there already use cases in mind for other invocations of `cached.R`? Perhaps it would be beneficial to have shorter alias to create this "Replaceable" cache since it's used so often.

Reply:
Two ways forward:
1. ```golang
   replaceable := cached.Atomic{}
   foo.WireInto(LastSuccess(replaceable))
   replaceable.Store(something)
   ```
2. introduce a custom LastSuccess variant `Replaceable` implementing data type.

Reply:
No strong preferences, 2 would be slightly more convenient from the perspective of a user who would call this function a bunch.

Reply:
have done 2 now. I like it.